test/Motor.Extensions.Hosting.Kafka_IntegrationTest/KafkaExtensionTests.cs

```diff
@@ -279,9 +278,126 @@ public async Task Consume_CriticalFailure_ApplicationIsStopped()
         await taskCompletionSource.Task;
         // Give consumer enough time to handle returned ProcessedMessageStatus
         await Task.Delay(TimeSpan.FromSeconds(1));
-        fakeLifetimeMock.Verify(mock => mock.StopApplication(), Times.Once);
+        _fakeLifetimeMock.Verify(mock => mock.StopApplication(), Times.Once);
+    }
+
+    [Fact(Timeout = 50000)]
+    public async Task Consume_CommitsEveryCommitPeriod()
+    {
+        var config = GetConsumerConfig<string>(_topic, maxConcurrentMessages: 1, retriesOnTemporaryFailure: 1);
+        config.CommitPeriod = 2;
+        config.AutoCommitIntervalMs = null;
+
+        using var consumer = GetConsumer(_topic, config, _fakeLifetimeMock.Object);
+        consumer.ConsumeCallbackAsync = CreateConsumeCallback(ProcessedMessageStatus.Success, _consumedChannel);
+
+        var cts = new CancellationTokenSource();
+        await consumer.StartAsync(cts.Token);
+        var execution = consumer.ExecuteAsync(cts.Token);
+
+        await PublishAndAwaitMessages(_consumedChannel, 3);
+
+        await WaitForCommittedOffset(consumer, 3);
+
+        cts.Cancel();
+        await execution;
     }
 
+    private async Task PublishAndAwaitMessages(Channel<byte[]> channel, int count)
+    {
+        for (var i = 0; i < count; i++)
+        {
+            await PublishMessage(_topic, "someKey", Message);
+        }
+
+        for (var i = 0; i < count; i++)
+        {
+            Assert.Equal(_expectedMessage, await channel.Reader.ReadAsync(CancellationToken.None));
+        }
+    }
+
+    /// <param name="consumer">Consumer to get offset from</param>
+    /// <param name="expectedOffset">Offset of last committed message + 1, starts at 0</param>
+    private async Task WaitForCommittedOffset<TData>(KafkaMessageConsumer<TData> consumer, long expectedOffset)
+    {
+        while (true)
+        {
+            var offsets = consumer.Committed();
+            var offset = offsets.FirstOrDefault()?.Offset;
+
+            _output.WriteLine($"Waiting for offset {expectedOffset} got {offset}");
+            if (offset == expectedOffset)
+            {
+                return;
+            }
+
+            await Task.Delay(10, CancellationToken.None);
+        }
+    }
+
+    [Fact(Timeout = 50000)]
+    public async Task Consume_CommitsEveryAutoCommitIntervalMs()
+    {
+        var config = GetConsumerConfig<string>(_topic, maxConcurrentMessages: 1, retriesOnTemporaryFailure: 1);
+        config.AutoCommitIntervalMs = 1;
```

Comment:
(nitpick)
We only know that the auto commit interval is responsible for the commit happening here because the default value for `CommitPeriod` is higher than the 2 processed messages here, right?
I think that would be easier to understand if we would explicitly configure `CommitPeriod` here to _number of processed messages + 1_.

Having written that: I think these tests would be easier to read if they had a variable like `numberOfProcessedMessages` that than reused instead of directly writing the value 2.

Reply:
With the logic changed

(offset + 1)  % CommitPeriod == 0

it's easier to understand and self-documenting.

Will also add a numberOfProcessedMessages.